back-end/GPTapi.js

```diff
@@ -0,0 +1,34 @@
+import { config } from "dotenv";
+import { Configuration, OpenAIApi } from "openai";
+import express from "express";
+import cors from "cors";
+
+config();
+
+const PORT = 8080;
+const app = express();
```

Comment:
It's good that you got it working with an Express server.
If needed, there are also 2 other servers that you can use which are already installed in this repo:
1. [NextJS's NodeJS server,](https://nextjs.org/docs/pages/building-your-application/routing/api-routes) which is defined inside of `front-end/pages/api`.
2. The `back-end/server.py` runs a Python Flask server
3. The actual MecSimCalc website (not in this repo) runs a Django server, which is very similar to Flask but just more files

Reply:
We're just experimenting with this right now so whatever works is fine.

Do note that all these servers are HTTP severs (express, flask, django). If you need a server that sends messages back and forth in real time, instead of one individual request at a time, then **websocket server** can be useful. Websockets keeps a connection alive the entire time.

You can look at 
- [socket.io](https://socket.io/) server
- [websocket ws](https://github.com/websockets/ws) [I think socket.io is easier to use than this one]

You may or may not need a websocket server, I dunno. But these are the links should you ned it